google/cloud/internal/status_utils.h

```diff
@@ -23,7 +23,9 @@ namespace cloud {
 GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
 namespace internal {
 
-bool IsClientOrigin(Status const& status);
+enum class ErrorOrigin { kUnknown, kClient };
+
+ErrorOrigin IsClient(Status const& status);
```

Comment:
`IsBlah()` still sounds like something to be used in a boolean context.

I would suggest reusing the same name.  That is,
```
ErrorOrigin ErrorOrigin(Status const&);
```
I believe you will find that is possible.  So, `if (ErrorOrigin(status) != ErrorOrigin::kClient) ...`.

[BTW, `IsClient()` doesn't appear to be used anywhere (except the test).  I presume that comes later?]

Reply:
Done. And yes, it will be used later.

Reply:
I was getting 

```
google/cloud/internal/status_utils.cc:23:1: error: 'ErrorOrigin' does not name a type
   23 | ErrorOrigin ErrorOrigin(Status const& status) {
      | ^~~~~~~~~~~
```

So renamed to GetErrorOrigin. Lmk what you think

Reply:
Sounds good.  Sorry for being misleading.

Reply:
All good. I think the point was valid either way. Thanks!